lib/std/Build/Step/Options.zig

```diff
@@ -41,135 +41,341 @@ pub fn addOption(self: *Options, comptime T: type, name: []const u8, value: T) v
 
 fn addOptionFallible(self: *Options, comptime T: type, name: []const u8, value: T) !void {
     const out = self.contents.writer();
+    try printType(self, out, T, value, 0, name);
+}
+
+fn printType(self: *Options, out: anytype, comptime T: type, value: T, indent: u8, name: ?[]const u8) !void {
     switch (T) {
         []const []const u8 => {
-            try out.print("pub const {}: []const []const u8 = &[_][]const u8{{\n", .{std.zig.fmtId(name)});
+            if (name != null) {
+                try out.print("pub const {}: []const []const u8 = ", .{std.zig.fmtId(name.?)});
```

Comment:
In places like this you can capture the payload of the optional:
```suggestion
            if (name) |some| {
                try out.print("pub const {}: []const []const u8 = ", .{std.zig.fmtId(some)});
```